setup.py

```diff
@@ -77,15 +95,76 @@ def finalize_options(self):
         ]
         super().finalize_options()
 
+    def add_optional_flags(self):
+        env = os.environ.copy()
+        if sys.platform == 'win32':
+            return env
+
+        cppflags = []
+        if 'CPPFLAGS' in os.environ:
+            cppflags.append(os.environ['CPPFLAGS'])
```

Comment:
Could CPPFLAGS just be passed as `extra_compile_args` instead?

Reply:
This is put in the environment for `SetupPackage.do_custom_build`, not distutils.